src/ImageCanvas/region-tools.js

```diff
@@ -47,10 +47,21 @@ export type Polygon = {|
   open?: boolean,
   points: Array<[number, number]>
 |}
+export type Circle = {|
+  ...$Exact<BaseRegion>,
+  type: "circle",
+  // radius: number,
+  // x and y indicate the coordinates of the centre of the circle
+  x: number,
+  y: number,
+  // xr and yr indicate the x and y coordinates of the current mouse pointer while dragging.
+  xr: number,
+  yr: number
```

Comment:
I'm lukewarm on this definition because technically in a circle `xr=yr` so it should really just have `r`. My suggestion is to have an optional `r?: number`, which is set in the case where `xr=yr` to be `r=xr=yr` because this is more consistent with expectation when dealing with circles.

Reply:
@seveibar, I started with a single r as you are suggesting but, since we are transforming the coordinates based on image height and width, single r won't be enough. 

In the current code, xr and yr aren't actually representing the radius exactly but are representing the coordinates to last point while dragging by a mouse.

Reply:
gotcha